scripts/python/slim_vcf2fasta_chrom.py

```diff
@@ -32,31 +31,56 @@ def args():
 
     args = parser.parse_args()
 
-    return args.vcf, args.table, args.region, \
-        args.mut_mat, args.outfile
+    return args.vcf, args.fasta, args.region, args.mut_mat, args.outfile
+
 
+def prep_samples(vcf: str, fasta: str, region: str): -> list, dict
+    """Prepare samples and reference sequences to populate with alt alleles.
 
-def prep_samples(vcf, table, region):
+    Args:
+        vcf: a bgzipped and tabixed SLiM VCF output file.
+        fasta: a FASTA containing the reference genome (or at minimum, 
+            the reference chromosome(s) of interest)
+        region: samtools formatted region to extract from reference
+            FASTA.
+
+    Returns:
+        - A list with phased sample names.
+        - A dictionary with phased sample names as keys and the desired
+              reference sequence as values.
+    """
     print('prepping samples...')
     vcf_in = VCF(vcf)
     samples = vcf_in.samples
     samples_phased = []
-    p = ANT.Reader(table)
+    # read in reference sequence
+    print('reading in reference sequence...')
     chrom, coords = region.split(':')
     start, end = [int(n) for n in coords.split('-')]
     seq_length = (end - start) + 1
-    ref_seq = ''.join(record.ref for record in tqdm(p.fetch(chrom, start-1, end)))
+    for record in SeqIO.parse(fasta, 'fasta'):
+        if chrom in str(record.id):
+            ref_seq = str(record.seq)[start-1:end]
+            break
```

Comment:
Seem like there should be a more efficient way of retrieving just a single record rather than looping though the reference. Not 100% sure, but what about an adapted version of something like this from [their documentation](https://biopython.org/wiki/SeqIO):

```
from Bio import SeqIO
record_dict = SeqIO.index("example.fasta", "fasta")
print(record_dict["gi:12345678"])  # use any record ID
```
I'm not sure what more efficient or even if it matters. Just a thought.

Reply:
I'd never seen this method before, but although it seems cleaner, a quick check shows that it's actually a lot slower than the loop:

```python
>>> import timeit
>>> test1 = '''from Bio import SeqIO
  1 fname = 'dmel-reference-r6.32.fasta'
  2 for record in SeqIO.parse(fname, 'fasta'):
  3     if '3L' in str(record.id):
  4         rec = str(record.seq)
  5        break
  6 '''
>>> test2 = '''from Bio import SeqIO
  1 fname = 'dmel-reference-r6.32.fasta'
  2 record_dict = SeqIO.index(fname, 'fasta')
  3 rec = str(record_dict['3L'])
  4 '''
>>> timeit.timeit(test1, number=100)
21.682134576141834
>>> timeit.timeit(test2, number=100)
53.35465686023235
```

This is probably because this is a smaller fasta as far as the number of records goes? Not too sure about the inner workings of `.index`.

Reply:
Hmm interesting. I assumed it would be faster but all good. Merging!